src/components/screens/DocsScreen/DocsSearch.tsx

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable react/require-default-props */
+import React from 'react';
+import { css, Global } from '@storybook/theming';
+import { global, styles } from '@storybook/design-system';
+import { DocSearch } from '@docsearch/react';
+import useSiteMetadata from '../../lib/useSiteMetadata';
+import '@docsearch/css';
+
+const { GlobalStyle } = global;
+
+const ALGOLIA_API_KEY = process.env.GATSBY_ALGOLIA_API_KEY;
+
+interface DocsSearchProps {
+  framework: string;
+  /** Only used for Storybook */
+  visible?: boolean;
+}
+
+export const classNames = {
+  BUTTON: '.DocSearch-Button',
+  ICON: '.DocSearch-Button .DocSearch-Search-Icon',
+  PLACEHOLDER: '.DocSearch-Button-Placeholder',
+};
+
+const idleButtonStyles = css`
+  background: ${styles.color.lightest};
+  box-shadow: ${styles.color.border} 0 0 0 1px inset;
+  color: ${styles.color.mediumdark};
+`;
+
+// Designed to match Input from @storybook/design-system
+const docSearchStyles = css`
+  ${classNames.BUTTON} {
+    ${idleButtonStyles}
+    font-size: ${styles.typography.size.s2}px;
+    line-height: 16px;
+    margin: 0;
+    padding: 10px 0 10px 14px;
+
+    &:focus,
+    &:focus:hover {
+      ${idleButtonStyles}
+    }
+
+    &:hover {
+      ${idleButtonStyles}
+    }
+  }
+
+  ${classNames.ICON} {
+    color: currentColor;
+    height: 1em;
+    margin-right: 8px;
+    width: 1em;
+  }
+
+  ${classNames.PLACEHOLDER} {
+    font-size: 1em;
+    padding: 0;
+  }
+
+  .DocSearch-Button-Keys {
+    display: none;
+  }
+
+  .DocSearch-Modal {
+    --docsearch-primary-color: ${styles.color.secondary};
+  }
+`;
+const DocSearchStyles = () => <Global styles={docSearchStyles} />;
+
+const label = 'Search docs';
```

Comment:
You could mention the keyboard shortcut here:
 
```suggestion
const label = 'Search docs (cmd +k)';
```

Reply:
Good call. Maybe in a tooltip on hover/focus?

Reply:
Or even just in the placeholder 

<img width="477" alt="CleanShot 2022-02-22 at 19 25 15@2x" src="https://user-images.githubusercontent.com/42671/155242535-ee5693f0-7fa8-4ffe-9d9f-702b402be14f.png">

Reply:
We have a style for the "keyboard shortcut key" in Storybook that looks like this: 
<img width="217" alt="image" src="https://user-images.githubusercontent.com/263385/155360116-4c4927ef-cc3f-4b5d-a3b3-7fe2b08cb895.png">

<img width="253" alt="image" src="https://user-images.githubusercontent.com/263385/155360257-7139ac79-3e93-4bd2-816d-edb468d695ab.png">


I believe there's a component for this too!

Reply:
I'll look into it!

Reply:
Boom:

![image](https://user-images.githubusercontent.com/486540/155619845-e69ae070-d8e7-4fbb-a5c6-e0c1001d9221.png)

There was no shared component, but I found the one used in Storybook's UI and copied it.

----

As an aside, I'm not sure how much sense it makes to use <kbd>^</kbd> to represent <kbd>Control</kbd> for non-Mac users. As I understand it, that's only a Mac convention. That translation comes from Storybook, though, so I kept it as-is to be consistent, and if we want to change it, we should do so over there.

Reply:
looks good to me 💪